src/xbot/common/subsystems/BaseSRXControlledSubsystem.java

```diff
@@ -0,0 +1,180 @@
+     * @param factory The WPIFactory
+     * @param pidPropertyManager The default PIDF values the system should use
+     * @param propManager The XPropertyManager
+     */
+    public BaseSRXControlledSubsystem(
+            String systemName,
+            int masterChannel,
+            boolean invertMaster,
+            boolean invertMasterSensor,
+            WPIFactory factory, 
+            PIDPropertyManager pidPropertyManager,
+            XPropertyManager propManager){
+        log.info("Creating " + systemName + " system");
+        
+        this.pidPropertyManager = pidPropertyManager;
+        this.systemName = systemName;
+        this.masterChannel = masterChannel;
+        
+        // Reading through the manual, it looks like only a few things 
+        // support human units. It might be easier to just keep everything in native units per 100ms,
+        // which is what the Talon uses for all its calculations.
+        // I call this Ticks per Deciseconds, or TPD
+        systemSpeedThresh = propManager.createPersistentProperty(systemName + " nominal speed thresh (TPC)", 1);
+        systemCurrentSpeed = propManager.createEphemeralProperty(systemName + " current speed (TPC)", 0);
+        systemTargetSpeed = propManager.createEphemeralProperty(systemName + " goal speed (TPC)", 0);
+        systemOutputPower = propManager.createEphemeralProperty(systemName + " voltage", 0);
+        atSpeedProp = propManager.createEphemeralProperty("Is" + systemName + " at speed?", false);
+        systemTalonError = propManager.createEphemeralProperty(systemName + " speed error", 0);
+        enablesystemLogging = propManager.createEphemeralProperty("Is " + systemName + " logging enabled?", false);
+        
+        masterMotor = factory.getCANTalonSpeedController(masterChannel);
+
+        initializeMotorConfiguration(invertMaster, invertMasterSensor);
+        masterMotor.createTelemetryProperties(systemName + "  master", propManager);
+    }
+    
+    /**
+     * Setup extra values for the masterMotor XCANTalon, and do any necessary setup of your
+     * follower XCANTalon(s).
+     * Some examples:
+     * -Configure your follower XCANTalon(s) to follow masterMotor
+     * -Configure voltage settings, such as configNominalOutputVoltage and configPeakOutputVoltage
+     */
+    protected abstract void setupExtendedMotorProperties();
+    
+    protected void initializeMotorConfiguration(boolean motorInverted, boolean motorSensorInverted) {
+        masterMotor.setFeedbackDevice(FeedbackDevice.QuadEncoder);
+        masterMotor.setInverted(motorInverted);
+        masterMotor.reverseSensor(motorSensorInverted);
+        masterMotor.setControlMode(TalonControlMode.Speed);
+        masterMotor.setProfile(0);
+        
+        setupExtendedMotorProperties();
+    }
+        
+    private void updateMotorPidValues() {
+        masterMotor.setP(pidPropertyManager.getP());
```

Comment:
ah nvm, just noticed where it's called and that it's private, alls well. Maybe would be fine to be protected though in case someone needs to tweak it?